### PR TITLE
Feature prometheus servlet, which exposes solr metrics in prometheus format  

### PR DESCRIPTION
**Pull request by hiteshk25**
[CH-214504]

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java [hunk 1]
```diff
@@ -0,0 +1,1003 @@
...
+    AtomicInteger qTime = new AtomicInteger();
+    for (MetricsApiCaller caller : callers) {
+      caller.call(qTime, metrics, request);
+    }
+    getSharedCacheMetrics(metrics, getSolrDispatchFilter(request).getCores(), cacheMetricTypes);
+    metrics.add(
+        new PrometheusMetric(
+            "metrics_qtime", PrometheusMetricType.GAUGE, "QTime for calling metrics api", qTime));
+    response.setCharacterEncoding(StandardCharsets.UTF_8.name());
+    PrintWriter printWriter =
```
Comment (hiteshk25) [inline solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java:99]: @justinrsweeney updated how we get printwriter

File: solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java [hunk 2]
```diff
@@ -0,0 +1,1003 @@
...
+
+    private final String queryString;
+    private final Map<String, Object> attributes = new HashMap<>();
+
+    MetricsApiRequest(HttpServletRequest request, String group, String prefix, String property)
+        throws IOException {
+      super(request);
+      queryString =
+          String.format(
+              Locale.ENGLISH,
```
Comment (hiteshk25) [inline solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java:765]: added locale

**Comment by justinrsweeney** [comment]
I think we could remove the `[LONG]` from the title and just replace the default description text with a short description of the feature and then merge

**Review by justinrsweeney — APPROVED** [review]
LGTM!

**Pull request merged**